metrics.go

```
@@ -28,18 +31,32 @@ func New(cfg Config) (*StatsReporter, error) {
 		cfg.Namespace = fmt.Sprintf("%s.", cfg.Namespace)
 	}
 
-	address := fmt.Sprintf("%s:%d", cfg.StatsdHost, cfg.StatsdPort)
+	var client metricsClient
+	if cfg.Lambda {
+		lambda := lambda.New(cfg.LambdaLogger)
```

Do you want to return an error here if the logger isn't defined? (ie, if i set `Lambda: true` without providing a logger)

---

Ah, good call!